pkg/plugin/probes-handler/probes_handler_test.go

```diff
@@ -0,0 +1,64 @@
+package probeshandler_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+
+	probeshandler "github.com/arquillian/ike-prow-plugins/pkg/plugin/probes-handler"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Test liveliness and readiness probes.", func() {
+
+	const (
+		probesEndpoint = "/version"
+		version        = "xxxxxxxx-xxxxxxxxxx"
+	)
+
+	var _ = BeforeSuite(func() {
+		os.Setenv("VERSION", version)
+	})
+
+	var _ = AfterSuite(func() {
+		os.Clearenv()
+	})
+
+	Context("When in healthy state", func() {
+		It("should return plugin version in response body", func() {
+			// given
+			probesHandler := probeshandler.NewProbesHandler()
+			request := httptest.NewRequest("GET", probesEndpoint, nil)
+			response := httptest.NewRecorder()
+			expectedBody := probeshandler.Probe{Version: version}
+
+			// when
+			http.Handle(probesEndpoint, probesHandler)
+			probesHandler.ServeHTTP(response, request)
+
+			// then
+			actualBody := probeshandler.Probe{}
+			json.Unmarshal(response.Body.Bytes(), &actualBody)
+			Expect(actualBody).To(Equal(expectedBody))
+		})
+	})
+
+	Context("When in unhealthy state", func() {
+		It("should return empty response body", func() {
+			// given
+			handler := func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusBadGateway)
+			}
+			request := httptest.NewRequest("GET", probesEndpoint, nil)
+			response := httptest.NewRecorder()
+
+			// when
+			handler(response, request)
+
+			// then
+			Expect(response.Body.String()).To(Equal(""))
```

Comment:
if I understand it correctly, when it fails/is unhealthy then it returns an empty string. is this really expected? what was behind this decision?
looking here https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/
there is written
> Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure.

this would mean that even when the proble is broken down then it would return 200.

Reply:
Here, my intention was to forcefully fail and check for the unhealthy state. 
> Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure.

As mentioned here, I add a dummy handler that returns the error code 502 that exceeds the success code for the probe forcing it to fail as can be seen in the output when tested manually.

```
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath			Type		Reason			Message
  ---------	--------	-----	----			-------------			--------	------			-------
  1m		1m		1	default-scheduler					Normal		Scheduled		Successfully assigned test-keeper-4-ggj4z to localhost
  1m		1m		1	kubelet, localhost					Normal		SuccessfulMountVolume	MountVolume.SetUp succeeded for volume "plugins" 
  1m		1m		1	kubelet, localhost					Normal		SuccessfulMountVolume	MountVolume.SetUp succeeded for volume "oauth" 
  1m		1m		1	kubelet, localhost					Normal		SuccessfulMountVolume	MountVolume.SetUp succeeded for volume "sentry-dsn" 
  1m		1m		1	kubelet, localhost					Normal		SuccessfulMountVolume	MountVolume.SetUp succeeded for volume "hmac" 
  1m		1m		1	kubelet, localhost					Normal		SuccessfulMountVolume	MountVolume.SetUp succeeded for volume "default-token-h8t5s" 
  1m		1m		1	kubelet, localhost	spec.containers{test-keeper}	Normal		Pulling			pulling image "172.30.1.1:5000/myproject/test-keeper:6d586fe-1524764442"
  1m		1m		1	kubelet, localhost	spec.containers{test-keeper}	Normal		Pulled			Successfully pulled image "172.30.1.1:5000/myproject/test-keeper:6d586fe-1524764442"
  1m		1m		1	kubelet, localhost	spec.containers{test-keeper}	Normal		Created			Created container
  1m		1m		1	kubelet, localhost	spec.containers{test-keeper}	Normal		Started			Started container
  1m		1m		3	kubelet, localhost	spec.containers{test-keeper}	Warning		Unhealthy		Liveness probe failed: HTTP probe failed with statuscode: 502
  1m		9s		11	kubelet, localhost	spec.containers{test-keeper}	Warning		Unhealthy		Readiness probe failed: HTTP probe failed with statuscode: 502
``` 
In this case, hitting the `/version` end-point returns empty value. Covering this case with the tests was my intention. 
Not sure, if this was the best approach to test it and I can update it if you can suggest any better way.